c/progress/manager.h

```diff
@@ -43,9 +44,10 @@ class progress_manager {
     // while more and more top-level tasks are received.
     progress_bar* pbar;
     std::stack<work*> tasks;
+    std::mutex mutex;
```

Comment:
Could you please add a comment about the purpose of this mutex: what it's supposed to protect, otherwise it's rather hard to verify that it is used in all places where its supposed to and not anywhere else.

Reply:
Sure.

Reply:
Done.